### DISCUSSION TITLE
Path for progammatically trimming volumes?

### DISCUSSION THREAD

**Discussion by LP0101** [longhorn/longhorn#5379] [Q&A] [answered]
I'm currently trying to set up a job that will automatically trim all my longhorn volumes.

According to the docs [here](https://longhorn.io/docs/1.4.0/volumes-and-nodes/trim-filesystem/), this can be done by running `fstrim <mountpath>`. However, I'm not sure what the mountpath actually is.

If I try running it from inside the pod that's mounting the volume, I get hit with `FITRIM ioctl failed: Operation not permitted`. Where do I actually run the fstrim on?

**Comment by weizhe0422** [comment, marked as answer]
The mount path will include two kind of mount point for volume provision: k8s global mount point and pod mount point. 

You can get the list with this command in your cluster node: `cat /proc/mounts | grep <longhorn-volume-device-path> | awk '{print $2}'`  and the result will be like this 
```
/var/lib/kubelet/plugins/kubernetes.io/csi/driver.longhorn.io/30f95cb2c541af6555af122fdd2dc7ee24322f15599b8fab5f21bc8f75d8c850/globalmount
/var/lib/kubelet/pods/0b5282a1-802e-4653-b0f7-d8cdfa15eeb0/volumes/kubernetes.io~csi/pvc-30317456-a35b-43be-b17e-ad000212cb11/mount
```

**Comment by weizhe0422** [comment]
The error message `FITRIM ioctl failed: Operation not permitted` seems like the device has noe TRIM permission. 

I think you can use `lsblk --discard` to list the devices that support TRIM and add `discard` in `/etc/fstab` to enable it.
